stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt

```diff
@@ -0,0 +1,335 @@
+        val chatClient: ChatClient = mock()
+        val channelClient: ChannelClient = mock()
+
+        val viewModel = Fixture(chatClient, channelClient)
+            .givenCurrentUser()
+            .givenChannelsQuery()
+            .givenChannelsState(
+                channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                loading = false
+            )
+            .givenChannelMutes()
+            .givenHideChannel()
+            .get()
+
+        viewModel.hideChannel(channel1)
+
+        verify(chatClient).hideChannel(
+            channelType = channel1.type,
+            channelId = channel1.id
+        )
+    }
+
+    @Test
+    fun `Given channel list in content state and the current user is online When loading more channels Should load more channels`() =
+        runTest {
+            val nextPageRequest = QueryChannelsRequest(
+                filter = queryFilter,
+                querySort = querySort,
+                offset = 30,
+                limit = 30,
+            )
+
+            val chatClient: ChatClient = mock()
+
+            val viewModel = Fixture(chatClient)
+                .givenCurrentUser()
+                .givenChannelsQuery()
+                .givenChannelsState(
+                    channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                    nextPageRequest = nextPageRequest,
+                    loading = false
+                )
+                .givenChannelMutes()
+                .givenIsOffline(false)
+                .get()
+
+            val mockObserver: Observer<ChannelListViewModel.State> = mock()
+            viewModel.state.observeForever(mockObserver)
+
+            viewModel.onAction(ChannelListViewModel.Action.ReachedEndOfList)
+
+            val captor = argumentCaptor<QueryChannelsRequest>()
+            verify(chatClient, times(2)).queryChannels(captor.capture())
+
+            captor.firstValue.offset `should be equal to` 0
+            captor.secondValue.offset `should be equal to` 30
+        }
+
+    @Test
+    fun `Given channel list in content state and the current user is offline When loading more channels Should do nothing`() =
+        runTest {
+            val chatClient: ChatClient = mock()
+            val viewModel = Fixture(chatClient)
+                .givenCurrentUser()
+                .givenChannelsQuery()
+                .givenChannelsState(
+                    channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                    loading = false
+                )
+                .givenChannelMutes()
+                .givenIsOffline(true)
+                .get()
+
+            val mockObserver: Observer<ChannelListViewModel.State> = mock()
+            viewModel.state.observeForever(mockObserver)
+
+            viewModel.onAction(ChannelListViewModel.Action.ReachedEndOfList)
+
```

Comment:
Also, we can add the `removeObserver` here for avoiding unexpected behavior.

Reply:
Good suggestion.

Removed the observer at the end of the test.